taxonomy/api/v1/views.py

```diff
@@ -186,3 +187,38 @@ def get_queryset(self):
             queryset = queryset.filter(username=self.request.user.username)
 
         return queryset.all().select_related('current_job').prefetch_related('skills', 'future_jobs')
+
+
+class JobHolderUsernamesAPIView(APIView):
+    """
+    THis API takes a job id as input and returns a list of 100 usernames
+    of job holders after querying the most recent SkillsQuiz table entries.
+    This API is available only to the admin users.
+    """
+    permission_classes = [permissions.IsAdminUser]
+    throttle_scope = 'taxonomy-api-throttle-scope'
+
+    def get(self, request, job_id):
+        """
+        Example URL: GET https://discovery.edx.org/taxonomy/api/v1/job-holder-usernames/2/
+
+        Example Response:
+        {
+            "usernames": [
+                "user1",
+                "user2",
+                "user3",
+                "user4",
+                "user5",
+                // 95 more usernames
+            ]
+        }
+        """
+        job = get_object_or_404(Job, id=job_id)
+
+        related_usernames = SkillsQuiz.objects.filter(current_job=job).order_by('-created')
+        usernames = related_usernames.order_by('-username').values_list('username', flat=True).distinct()[:100]
```

Comment:
Here we are adding two `oder_by`. Second Order_by is overriding first `order_by`. So the raw query for above query is followings
```
'SELECT DISTINCT `taxonomy_skillsquiz`.`username` FROM `taxonomy_skillsquiz` WHERE `taxonomy_skillsquiz`.`current_job_id` = 1 ORDER BY `taxonomy_skillsquiz`.`username` DESC LIMIT 100'
```

That clearly shows there is no order based on the `created` field. As field-based distinct is not supported in MySQL, We have to get distinct records using python. here is example query that I have tested and working good.
```
from collections import OrderedDict
usernames_query = SkillsQuiz.objects.filter(
    current_job_id=1
).order_by(
    '-id'
).values_list(
    "username", flat=True
)[:5000]
usernames = list(OrderedDict.fromkeys(usernames_query))[:100]
```

Reply:
Note: we can use -id as an alternative of -created, id ordering should be fast as it is a primary key.

Reply:
@HammadAhmadWaqas I think this logic would not work in certain cases like if we have 5000 recent responses from the same user then would return just one user but that is highly unlikely. I have changed the usernames extraction logic as suggested.